### ISSUE TITLE
Incorrect @Schemable macro expansion in nested types.

### ISSUE THREAD

**Issue by esconner** [ajevans99/swift-json-schema#147]
I use DTOs for the model types in my project. The `@Schemable` macro generates incorrect code when DTOs are nested resulting in an initially misleading compiler error. Here's a contrived simplified example:
```swift
		public struct ModelElementType {
			var v1: Double
			var v2: Double
			
			@Schemable
			struct DTO: Codable {
				var v1: Double
				var v2: Double
			}
		}
		
		public struct ModelType {
			var name: String
			var value: ModelElementType
			
			@Schemable
			public struct DTO: Codable {
				var name: String
				var element: ModelElementType.DTO
			}
		}
```
The problem is the `element` property in `ModelType`. The macro expansion for `element` (`JSONProperty(key: "element") { JSONDynamicReference<Self>() }`) is obviously incorrect. The `Self` indicates that the macro logic thinks `element` is of type `ModelType.DTO` (aka `Self`) when it is actually `ModelElementType.DTO`. I think this may be because the logic that detects legitimate circular DTO references compares the unqualified names of the DTOs.

**Comment by ajevans99** [comment]
Good catch. Merged a fix
